[DOCS] Add data streams to API conventions (#58695)
elastic/elasticsearch: docs/reference/snapshot-restore/take-snapshot.asciidoc

// TEST[skip:cannot complete subsequent snapshot]

> The list of indices that should be included into the snapshot can be specified using the `indices` parameter that

Reviewer: ```suggestion
Use the `indices` parameter, which supports <<multi-index,multi-target syntax>>, to specify the list of indices to include in the snapshot.
```

Author: As this PR is primary concerned with the multi-index xref, I'm going to leave this as-is for now. It'll create fewer merge conflicts for me in #58513.

> supports <<multi-index,multi-target syntax>>, although the options which control the behavior of multi index syntax
> must be supplied in the body of the request, rather than as request parameters. The snapshot request also supports the

Reviewer: ```suggestion
You specify the options that control the behavior of multi-target syntax
in the body of the request, rather than as request parameters. The snapshot request also supports the
```

Author: Same as above.
